lib/engine/share_pool.rb

```diff
@@ -88,10 +88,8 @@ def buy_shares(entity, shares, exchange: nil, exchange_price: nil, swap: nil, al
       if price.zero?
         transfer_shares(bundle, entity)
       else
-        receiver = if corporation.capitalization == :incremental && bundle.owner.corporation? || bundle.owner.player?
-                     bundle.owner
-                   elsif corporation.capitalization == :escrow && bundle.owner.corporation?
-                     # When another game with escrow capitalization is implemented put this into @game
+        receiver = if %i[escrow incremental].include?(corporation.capitalization) && bundle.owner.corporation? ||
+                       bundle.owner.player?
```

Comment:
Oh wait the old code here actually has different behavior; if `corporation.capitalization` is `:escrow` and `bundle.owner` is a player, the old code would end up with `receiver = @bank` but the new code does `receiver = bundle.owner`

Reply:
Hm, no
old: bundle.owner.player? == true => receiver = bundle.owner
(false && false || true => true)

Reply:
Well this is a little weird but it makes rubocop happy 🤷 

```
receiver = if (%i[escrow incremental].include?(corporation.capitalization) && bundle.owner.corporation?) ||
              (corporation.capitalization == :incremental && bundle.owner.player?)
             bundle.owner
           else
             @bank
           end
```

Reply:
Seems a little buggy that the cop doesn't detect the `|| bundle.owner.player?` part as something that would make it a non-dup branch

Reply:
That’s wrong. If bundle.owner.player? == true and everything else is false, receiver = bank.
old code was different!

Reply:
> Seems a little buggy that the cop doesn't detect the `|| bundle.owner.player?` part as something that would make it a non-dup branch

It’s because the result is the same. That’s what’s checked.

Reply:
Oh didn't see your comment until after making my two replies here. Will think about this some more...

Reply:
Oh I see, `A && B || C` is `(A && B) || C`, I was thinking of it as `A && (B || C)` ☠️

Reply:
New suggestion for my own sake 😛 

```suggestion
        receiver = if (%i[escrow incremental].include?(corporation.capitalization) && bundle.owner.corporation?) ||
                       bundle.owner.player?
```

Reply:
> New suggestion for my own sake stuck_out_tongue

I despise unnecessary parentheses. Commit if you must ;-)

Reply:
haha my view is that parens which are unnecessary for the computer are way better than humans needing to know anything about operator precedence. I will do what I must 😁